app/controllers/application_controller.rb

```diff
@@ -6,16 +6,63 @@ def log_in(user)
   end
 
   def log_out
+    forget(current_user)
     session.delete(:user_id)
     @current_user=nil
   end
 
+  def forget(user)
+    user.forget
+    cookies.delete(:user_id)
+    cookies.delete(:remember_token)
+  end 
+
+
+  helper_method :current_user
+  def current_user
+    if (user_id = session[:user_id])
+      @current_user ||= User.find_by(id: user_id)
+    elsif (user_id = cookies.signed[:user_id])
```

Comment:
Though I haven't actually tested this yet (and am skeptical), the a user's id should not persist within the value for session[:user_id] upon browser close. In the case that a user closes their browser and revisits the site having ticked off remember me (thus enabling cookies), the current user method in my current implementation should not enter into the first if statement of the `current_user` method because the condition would evaluate to `false`. In that case, the elsif block would be executed because the value for cookies[:user_id] should theoretically not be nil. Sorry, if I'm not understanding your question exactly and just stated something completely obvious and irrelevant. Thanks.